runtime/v2/shim/util_unix.go

```diff
@@ -99,10 +99,13 @@ func NewSocket(address string) (*net.UnixListener, error) {
 	if err != nil {
 		return nil, err
 	}
-	if err := os.Chmod(path, 0600); err != nil {
-		os.Remove(sock.path())
-		l.Close()
-		return nil, err
+
+	if !sock.isAbstract() {
```

Comment:
What about calling `isAbstract := sock.isAbstract()` at beginning and reuse the result here?

Reply:
Hi @fuweid 

Updated.